### ISSUE TITLE
has youtube blocked xdm?

### ISSUE THREAD

**Issue by a1damF** [subhra74/xdm#946]
## Checklist

- [ ] I'm asking a question
- [ ] I've looked through the README and FAQ for similar questions
- [ ] I've searched the bugtracker for similar questions including closed ones

## Question

good morning all,

tried to download some youtube clips yesterday and today but to no avail....
xdm returns this message: XDM could not download the file as server sent an invalid response.

i'm running xdm 7.2.11,  linux mint 21 with the latest kernel on two laptops and firefox 107.0

yesterday i reinstalled xdm, same version, still no change.

xdm, however works alright on vimeo and dailymotion, only it doesn't on youtube.

i also installed video DownloadHelper, just to see if it would work with youtube and it didd. however it was very slow and i never really like this downloader so i uninstalled it....

anyway, by the look of it it seems to me that youtube must have blocked xdm, any of you guys has the same problems?

cheers,
adam

**Comment by imnotadev-ineedhelp** [comment]
yep i'm having same issue, youtube downloading is not working, for some reason :(

**Comment by a1damF** [comment]
i have a funny feeling that what's causing problems could be the latest version of firefox, can't proove it though....
i've just tried to download some clips from youtube - can download some, can't download others.
also, some clips can be downloaded only after refreshing youtube page which is strange....
anyway, i guess i will have to wait it out for a while and see what happens in some days/weeks.
cheers

**Comment by a1damF** [comment]
just a quick update....

just upgraded firefox from 107.0 to 107.0.1 and had a try at downloading clips from youtube....
no problem at all, could download every clip i wanted to without any hassles....

one thing though, all clips that xdm downloads have the same generic name "videoplayback.mp4" but that can probably be sorted out later, can live with that when downloading just one clip at a time (can copy and paste the title) but this could be a bit of a hassle when downloading several clips at the same time.....

anyway, you guys have a good day,
cheers,
adam

**Comment by billypoon34-hello** [comment]
Still not work on YT even update firefox to 107.0.1 and retry or using chrome
The chrome version only download the video name but content is YT advertisement

**Comment by billypoon34-hello** [comment]
Just change to use this 
https://github.com/KurtBestor/Hitomi-Downloader/wiki/Scripts
speed also same as XDM

**Comment by Shazam72** [comment]
> I have  firefox v108 but i get the same issue as well before update

**Comment by billypoon34-hello** [comment]
The hitomi is using goodbyedpi
for passing the security of youtube website block
That to success continue to download the videos

**Comment by Shazam72** [comment]
> Guys, I am getting this "XDM could not download the file as server sent an invalid response." when I try to download any video using firefox but I have just tried to download youtube videos using chrome, it's working fine!

I should try on my own

**Comment by Shazam72** [comment]
@realmzakaria i tried with chromium and its work well. I used an older version of xdman chromium plugin. But there is a warning that the manifest version used in the plugin is depreciated and will be soon removed. Screenshots below
![Screenshot_1](https://user-images.githubusercontent.com/49822437/208150073-ac788dd5-8121-4df9-8c98-35e06ce667d4.png)
![Screenshot_2](https://user-images.githubusercontent.com/49822437/208150088-f662f3d3-238a-47cf-9d29-d6e921ee3498.png)

**Comment by Shazam72** [comment]
Its really weird that Chrome(Chromium) allow yt download but firefox doesn´t. That should be a problem with Firefox

**Comment by smestarz** [comment]
Till few weeks ago I was able to download youtube videos using XDM. but then after few updates of firefox the XDM stopped working (to download videos) I am using Firefox 108.0.1 and still unable to download/
I also use Google Chrome, and XDM works flawlessly on chrome. so the issue is not Youtube, but maybe firefox itself. as without changing anything, I can use XDM on google chrome to download youtube videos, so the issue is firefox

**Comment by watsn815263** [comment]
Youtube  files used to be labeled .mkv but as of 2 weeks ago they are now .webm files.  Competing products ARE working with FireFox so XDAM is at fault for not recognizing the .webm files.  This needs to be addresses asap because the user base of XDAM will disappear. (Video Download Helper and Flairget work fine) in Firefox.

**Comment by watsn815263** [comment]
> Its really weird that Chrome(Chromium) allow yt download but firefox doesn´t. That should be a problem with Firefox

see my post

**Comment by CyberTeckyLLC** [comment]
Hello,

I'm not sure if this applies to your situation, but when I'm using XDM with Firefox running on Windows 10 I don't seem to have any problem.  Although, I did notice that I needed to download/install [2] specific XDM applications to achieve the perfect outcome, which are:

1) Xtreme Download Manager (.msi) installed and running. [Link<https://xtremedownloadmanager.com/#downloads>]

2) XDM Browser Monitor for Firefox Extension - Installed and running [Link<https://addons.mozilla.org/en-US/firefox/addon/xdm-browser-monitor/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search>]

[Clarification] Result of installing/running both 'Xtreme Download Manager' and 'XDM Browser Monitor' extension should produce a new small floating "DOWNLOAD VIDEO" box that appears at the bottom far-right corner of the display as a result of XDM auto-detecting media on a given webpage.

3) Once you arrive on the target webpage containing the desired video media such as, 'YouTube.com' then look for that "DOWNLOAD VIDEO" floating box, then simply click the ONLY option available which probably looks like this:

Example

[null ###.# MB] Name of Video...

4) Finally, after clicking the option previously mentioned above, a new pop-up should appear that is called/titled, "VIDEO DOWNLOAD".  Simply select the 'Convert to' drop down option which should offer a multitude of media format solutions to choose and download from.

Hope this helps!

~ CyberTecky

**Comment by Damilarondo** [comment]
Well as I was looking at this issue trying to get solution I mistakenly press the Firefox extension link and I decided to scroll to the bottom of the page and I saw another extension by the name XDM integration Module by the same publisher and it had been updated 9 days ago (as from date of comment) so I installed it but it did not recognize my XDM browser monitor even after I restarted both Firefox and XDM, so I concluded that the extension must have been for a newer version of XDM and searched, I discovered an experimental beta link on the XDM's Git, installed it after seeing it was updated 3 weeks ago (as from date of comment), well it worked with the newer extension and downloads YouTube videos even faster than the previous version. The link is https://github.com/subhra74/xdm-experimental-binaries/tags

**Comment by imnotadev-ineedhelp** [comment]
@Damilarondo 
for some reason, on my side xdm experimental is not working, did downloaded [this](https://github.com/subhra74/xdm-experimental-binaries/releases/tag/8.0.14-beta) version.

**Comment by ariful-roni** [comment]
> > I have  firefox v108 but i get the same issue as well before update

same problem!

**Comment by watsn815263** [comment]
I'm totally lost on this thread.  mp4 - d/l FINE - mkv files on Youtube are NOT downloading.  Close to deleting everything and go to something else.  mkv files were d/l fine up until early December 2022

    On Sunday, December 25, 2022 at 04:55:48 AM CST, imnotadev-ineedhelp ***@***.***> wrote:  
 
 


@Damilarondo
for some reason, on my side xdm experimental is not working, did downloaded this version.

**Comment by brumbie** [comment]
Download the experimental binary;
[(https://github.com/subhra74/xdm-experimental-binaries/tags)]
Download the xdm-integration-module from firefox add ons.
Install the new exp binary and tick media streaming downloads. This will install an extra app from github.
Disable the old XDM Browser Monitor add on on Firefox.
Enable the new xdm-integration-module add on on Firefox.

Everything seems to be working again.

**Comment by ariful-roni** [comment]
Thank you for your suggestion!❤️

**Comment by watsn815263** [comment]
I am starting with a clean install on a new harddrive.  Using Linux (Debian based).  
Do I start with installation of XDM 2020 Version 7.2.11 package first and then do 
sudo xdman_gtk_8.0.18_amd64.deb?  I will wait for a response before I install becausethis program is difficult to uninstall. 
Thank you



  
    On Thursday, January 12, 2023 at 12:01:16 AM CST, brumbie ***@***.***> wrote:  
 
 


Download the experimental binary;
[(https://github.com/subhra74/xdm-experimental-binaries/tags)]
Download the xdm-integration-module from firefox add ons.
Install the new exp binary and tick media streaming downloads. This will install an extra app from github.
Disable the old XDM Browser Monitor add on on Firefox.
Enable the new xdm-integration-module add on on Firefox.

Everything seems to be working again.
